### DISCUSSION TITLE
[FeedBack] Student hoping for feedback on a few contracts

### DISCUSSION THREAD

**Discussion by jotaAfonso** [vyperlang/vyper#3210] [Q&A]
Hello, I'm a student currently doing a thesis, involving smart contracts, and I would like to receive some feedback on a few contracts that I made.

Contracts:

- https://github.com/jotaAfonso/Smart_Contracts/tree/main/vyper

The contracts follow a state machine approach.
The automaton that corresponds to the logic of the contract is in the README file.

There are a few of them, but I would like to highlight **two specific contracts**: 
- the **bazaar** because it is a contract that interacts with another contract;
- and the **auction** because there is more logic than the rest of the other contracts.

Looking for feedback on glaring errors and if there are some improvements to be made.

Thank you in advance and best wishes.

**Comment by fubuloubu** [comment]
You might have some better luck in the Vyper discord: https://discord.gg/xhE3qQSZ
